test/python/test_ibmqjob.py

```diff
@@ -214,42 +214,15 @@ def test_run_async_device(self):
         job_ids = [job.id for job in job_array]
         self.assertEqual(sorted(job_ids), sorted(list(set(job_ids))))
 
+    @slow_test
     def test_cancel(self):
-        if not self.using_ibmq_credentials:
-            self.skipTest('job cancellation currently only available on hubs')
-        backends = [backend for backend in self._provider.available_backends()
-                    if not backend.configuration['simulator']]
-        self.log.info('devices: %s', [b.name for b in backends])
-        backend = backends[0]
-        self.log.info('using backend: %s', backend.name)
-        num_qubits = 5
-        qr = QuantumRegister(num_qubits, 'qr')
-        cr = ClassicalRegister(num_qubits, 'cr')
-        qc = QuantumCircuit(qr, cr)
-        for i in range(num_qubits-1):
-            qc.cx(qr[i], qr[i+1])
-        qc.measure(qr, cr)
-        qobj = transpiler.compile(qc, backend)
-        num_jobs = 3
-        job_array = [backend.run(qobj) for _ in range(num_jobs)]
-        success = False
-        self.log.info('jobs submitted: %s', num_jobs)
-        while any([job.status['status'] == JobStatus.INITIALIZING for job in job_array]):
-            self.log.info('jobs initializing')
-            time.sleep(1)
-        for job in job_array:
-            job.cancel()
-        while not success:
-            job_status = [job.status for job in job_array]
-            for status in job_status:
-                self.log.info(status)
-            if any([status['status'] == JobStatus.CANCELLED for status in job_status]):
-                success = True
-            if all([status['status'] == JobStatus.DONE for status in job_status]):
-                raise IBMQJobError('all jobs completed before any could be cancelled')
-            self.log.info('-' * 20)
-            time.sleep(2)
-        self.assertTrue(success)
+        backend = self._provider.get_backend('ibmqx4')
```

Comment:
A final adjustments after merging #712, and we should be good to go! Can you make use of [`self.using_ibmq_credentials`](https://github.com/Qiskit/qiskit-terra/blob/master/test/python/common.py#L44) and use `ibmq_20_tokyo` as the backend if that is `True`? While we need to take a pass at the `@slow_test`s as a whole, it would help for running it against premium.